test/spec/modules/conversantAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,733 @@
+    });
+
+    it('isInt() should return true', function() {
+      expect(caObj.isInt(1)).to.equal(true);
+      expect(caObj.isInt(0)).to.equal(true);
+      expect(caObj.isInt(-1)).to.equal(true);
+      expect(caObj.isInt(111)).to.equal(true);
+      expect(caObj.isInt(1.0)).to.equal(true);
+    });
+
+    it('isInt() should return false', function() {
+      expect(caObj.isInt(0.1)).to.equal(false);
+      expect(caObj.isInt(1.1)).to.equal(false);
+      expect(caObj.isInt('foo')).to.equal(false);
+      expect(caObj.isInt(undefined)).to.equal(false);
+      expect(caObj.isInt(false)).to.equal(false);
+    });
+
+    it('initializeBidDefaults() should return correct object', function() {
+      let bid = caObj.initializeBidDefaults();
+      expect(Array.isArray(bid.eventCodes)).to.equal(true);
+      expect(bid.eventCodes.length).to.equal(0);
+      expect(Object.keys(bid).length).to.equal(1);
+    });
+
+    it('createAdUnit() should return correct object', function() {
+      let adUnit = caObj.createAdUnit();
+      expect(Array.isArray(adUnit.sizes)).to.equal(true);
+      expect(adUnit.sizes.length).to.equal(0);
+      expect(typeof adUnit.bids).to.equal('object');
+      expect(Object.keys(adUnit.bids).length).to.equal(0);
+      expect(Object.keys(adUnit).length).to.equal(2);
+    });
+
+    it('createAdSize() should return correct object', function() {
+      let adSize = caObj.createAdSize(1, 2);
+      expect(adSize.w).to.equal(1);
+      expect(adSize.h).to.equal(2);
+      expect(Object.keys(adSize).length).to.equal(2);
+
+      adSize = caObj.createAdSize();
+      expect(adSize.w).to.equal(-1);
+      expect(adSize.h).to.equal(-1);
+
+      adSize = caObj.createAdSize('foo', 'bar');
+      expect(adSize.w).to.equal(-1);
+      expect(adSize.h).to.equal(-1);
+    });
+
+    it('getLookupKey() should return correct object', function() {
+      let foo; // undefined var
+      let key = caObj.getLookupKey(foo, foo, foo);
+      expect(key).to.equal('undefined-undefined-undefined');
+
+      key = caObj.getLookupKey('foo', 'bar', 'baz');
+      expect(key).to.equal('foo-bar-baz');
+    });
+
+    it('createPayload() should return correct object', function() {
+      const REQUEST_TYPE = 'foo';
+      const AUCTION_ID = '124 abc';
+      conversantAnalytics.enableAnalytics(VALID_ALWAYS_SAMPLE_CONFIG);
+
+      let payload = caObj.createPayload(REQUEST_TYPE, AUCTION_ID);
+      expect(payload.requestType).to.equal(REQUEST_TYPE);
+      expect(payload.auction.auctionId).to.equal(AUCTION_ID);
+      expect(payload.auction.preBidVersion).to.equal(PREBID_VERSION);
+      expect(payload.auction.sid).to.equal(VALID_ALWAYS_SAMPLE_CONFIG.options.site_id);
+      expect(Object.keys(payload.auction).length).to.equal(3);
+
+      expect(typeof payload.adUnits).to.equal('object');
+      expect(Object.keys(payload.adUnits).length).to.equal(0);
+
+      expect(Object.keys(payload).length).to.equal(3);
```

Comment:
Probably can be done with deep equal again.  Check other tests below and replace with deep equal whenever it makes sense.